app/code/Magento/Catalog/Test/Unit/Block/Product/View/AttributesTest.php

```diff
@@ -0,0 +1,187 @@
+use \Magento\Catalog\Model\Product;
+use \Magento\Framework\View\Element\Template\Context;
+use \Magento\Framework\Registry;
+use \Magento\Framework\Pricing\PriceCurrencyInterface;
+use \Magento\Catalog\Block\Product\View\Attributes as AttributesBlock;
+
+/**
+ * Test class for \Magento\Catalog\Block\Product\View\Attributes
+ *
+ * @SuppressWarnings(PHPMD.LongVariable)
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class AttributesTest extends TestCase
+{
+    const TESTATTRIBUTENAME = 'phrase';
+    const TESTPHRASETEXTISBLANK = '';
+    const TESTPHRASETEXTNOTBLANK = 'Yes';
+
+    /**
+     * @var \Magento\Framework\Phrase
+     */
+    private $phraseResource;
+
+    /**
+     * @var \PHPUnit_Framework_MockObject_MockObject|\Magento\Eav\Model\Entity\Attribute\AbstractAttribute
+     */
+    private $attributeMock;
+
+    /**
+     * @var \PHPUnit_Framework_MockObject_MockObject|\Magento\Eav\Model\Entity\Attribute\Frontend\AbstractFrontend
+     */
+    private $frontendAttributeMock;
+
+    /**
+     * @var \PHPUnit_Framework_MockObject_MockObject|\Magento\Catalog\Model\Product
+     */
+    private $productMock;
+
+    /**
+     * @var \PHPUnit_Framework_MockObject_MockObject|\Magento\Framework\View\Element\Template\Context
+     */
+    private $contextMock;
+
+    /**
+     * @var \PHPUnit_Framework_MockObject_MockObject|\Magento\Framework\Registry
+     */
+    private $registryMock;
+
+    /**
+     * @var \PHPUnit_Framework_MockObject_MockObject|\Magento\Framework\Pricing\PriceCurrencyInterface
+     */
+    private $priceCurrencyInterfaceMock;
+
+    /**
+     * @var \Magento\Catalog\Block\Product\View\Attributes
+     */
+    private $attributesBlock;
+
+    protected function setUp() // @codingStandardsIgnoreLine
+    {
+        $this->phraseResource = new Phrase( // @codingStandardsIgnoreLine
+            self::TESTPHRASETEXTISBLANK
+        );
+
+        $this->attributeMock = $this
+            ->getMockBuilder(AbstractAttribute::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->attributeMock
+            ->expects($this::any())
+            ->method('getIsVisibleOnFront')
+            ->will($this::returnValue(true));
+
+        $this->attributeMock
+            ->expects($this::any())
+            ->method('getAttributeCode')
+            ->will($this::returnValue(self::TESTATTRIBUTENAME));
+
+        $this->frontendAttributeMock = $this
+            ->getMockBuilder(AbstractFrontend::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->attributeMock
+            ->expects($this::any())
+            ->method('getFrontendInput')
+            ->willReturn(self::TESTATTRIBUTENAME);
+
+        $this->attributeMock
+            ->expects($this::any())
+            ->method('getFrontend')
+            ->willReturn($this->frontendAttributeMock);
+
+        $this->productMock = $this
```

Comment:
I do not insist on changing all mocks creation this time, but generally `createMock` with all statements in one line would be much faster to read.

Reply:
Lines not exceeding 120 characters are per the MEQP2 standard available here:
https://github.com/magento/marketplace-eqp

Reply:
Sorry, I don't understand this comment.

Yeah, 120 characters limit should be honored, just there is no need to break the line for each method call, it could be:
```
$mock->expects()->method()
    ->willReturn()
```

Reply:
I structured it this way to avoid exceeding line length, but I also find it easier to read with a method on each line so I can read it like a checklist of items that are completed to get the final result. I can refactor if it is a big deal.

Reply:
It more a matter of taste, as I said, there is no strict requirement regarding this. Please leave as is.